### PR TITLE
Subaru: log eyesight faults as acc faults

### LINKED ISSUE #29404: Subaru: log ACC faults
- [ ] Figure out which of the four fault signals mean what, and which should we log

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/car/subaru/carstate.py [hunk 1]
```diff
@@ -19,6 +19,8 @@ def __init__(self, CP):
   def update(self, cp, cp_cam, cp_body):
     ret = car.CarState.new_message()
 
+    cp_es_distance = cp_body if not ((self.CP.flags & SubaruFlags.HYBRID) or (self.CP.flags & SubaruFlags.GLOBAL_GEN2)) else cp_cam
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:22]:
```suggestion
    cp_es_distance = cp_body if not (self.CP.flags & (SubaruFlags.HYBRID | SubaruFlags.GLOBAL_GEN2)) else cp_cam
```

**Review by sshane — APPROVED** [review]

File: selfdrive/car/subaru/carstate.py (hunk 1 above)
Comment (sshane) [reply selfdrive/car/subaru/carstate.py:22]: Wait, this is a duplicate of line 87

File: selfdrive/car/subaru/carstate.py [hunk 2]
```diff
@@ -29,6 +31,15 @@ def update(self, cp, cp_cam, cp_body):
       cp_brakes = cp_body if self.CP.flags & SubaruFlags.GLOBAL_GEN2 else cp
       ret.brakePressed = cp_brakes.vl["Brake_Status"]["Brake"] == 1
 
+    if not (self.CP.flags & SubaruFlags.HYBRID):
```
[the remark this answers is not in the document]
Comment (jnewb1) [reply selfdrive/car/subaru/carstate.py:34]: doesn't have ES_Distance (which is ES -> Engine). we still need a replacement for them to be upstreamed

https://github.com/commaai/openpilot/blob/master/selfdrive/car/subaru/interface.py#L15-L18

File: selfdrive/car/subaru/carstate.py (hunk 1 above)
Comment (jnewb1) [reply selfdrive/car/subaru/carstate.py]: yea meant to move it up

Comment (jnewb1) [reply selfdrive/car/subaru/carstate.py]: moved the other one up

File: selfdrive/car/subaru/carstate.py [hunk 3]
```diff
@@ -19,6 +19,8 @@ def __init__(self, CP):
   def update(self, cp, cp_cam, cp_body):
     ret = car.CarState.new_message()
 
+    cp_es_distance = cp_body if self.CP.flags & (SubaruFlags.GLOBAL_GEN2 | SubaruFlags.HYBRID) else cp_cam
+
```
Comment (sshane) [inline selfdrive/car/subaru/carstate.py:23]:
```suggestion
```

**Pull request merged**